roles/ceph-defaults/tasks/facts.yml

```diff
@@ -184,11 +184,11 @@
     ceph_uid: 64045
   when:
     - containerized_deployment
-    - ceph_docker_image_tag | search("latest") or ceph_docker_image_tag | search("ubuntu")
+    - ceph_docker_image | search("ubuntu")
```

Comment:
`ceph_docker_image` is either `ceph/daemon` or `rhceph` downstream so this won't work. My suggestion was to add an extra task that will do:

```
 - name: set_fact ceph_uid for Red Hat based system
   set_fact:
     ceph_uid: 167
   when:
     - containerized_deployment
     - ceph_docker_image | search("rhceph")
```

Reply:
Hey @leseb, after talking with some other folks about this; how would you feel about having `ceph_uid: 167` configured in the `all.yml` under Docker section? Then we could just keep only the debian task with the following conditionals: 
```
when:
  - containerized_deployment
  - ceph_uid is not defined
```
So basically, if you're running a RH family deployment ceph_uid will need to be set. Otherwise ceph_uid would get configured w/64045 if not defined. 

I think this is the best approach since we really have no way of controlling tags in other registries outside of docker.io/redhat.... which is what I think you were getting at.

Also, the task on 189 doesn't overwrite ceph_uid because ansible doesn't like the conditional on line 194 for some reason.... The or conditional I submitted uses the same var throughout and works fine.

Reply:
@utp887 we now use centos by default in our container image upstream so I'm sure if we should stick with 64045.
We don't really want to set the uid in group_vars because this will mean an extra config option. We can easily detect that.
If line 189 is not behaving properly then we need to fix that instead of changing the whole behavior.

If you want to override `ceph_uid` you can do so by running ansible with `-e ceph_uid=167` if your registry is not docker or redhat.

Do you mind running the test again with the line I suggested?
Thanks.